quinn-h3/src/tests/helpers.rs

```diff
@@ -12,6 +12,34 @@ use crate::{
     server::{self, IncomingConnection, Server},
 };
 
+#[macro_export]
```

Comment:
If the only difference between the zero- and one-argument forms is passing the argument to a nested function, maybe just make these function calls that take an `Option`? Macros seem like more complexity than needed here.

Reply:
I admit the YAGNI case here... I've been thinking about the uses-cases for this request construction features. I'd like to be able to pass headers for future tests, similarly with `serde`'s `json!()` macro:

```rust
get!("/path", { "name": "John Doe" })
post!("/path", body)
post!("/path", { "name": "John Doe" },  body)
```

But I was only focused on shortening tests. Do you think that doesn't worth it?

Reply:
I've been looking at stuff like this recently. Maybe it would be useful to just use the [`http::request::Builder`](https://docs.rs/http/0.2.0/http/request/struct.Builder.html) here, and add wrappers if some common cases are still not concise enough? It also has a relatively lightweight way to add headers.

Reply:
You are right, I got those macros out and it's fine. Let's write things when we need them.

I got the changes in the 0RTT impl commit. I also squashed the "more tests" commits because it was essentially the same subject (and that made the macros easier to remove).